src/reducers/index.js

```diff
@@ -511,17 +522,23 @@ export const resetSelectedTokenIfNeeded = (state, action) => {
   return state;
 };
 
-/*
+/**
  * Used when registering or creating tokens to update the wallet token list.
-*/
+ * @param {Record<string,string>} state.allTokens - Current list of allTokens
+ * @param {string} action.payload.uid - UID of token to add
+ */
 export const onNewTokens = (state, action) => {
-  // Add new created token to the all tokens set
-  state.allTokens.add(action.payload.uid);
+  // Convert `allTokens` to a Set to prevent duplicates
```

Comment:
```suggestion
  // Create a new object to prevent false negative on object comparisons
```

React will not trigger updates if the "dependency" (of hooks) do not change and object comparisons do not check contents, only that it is the same instance.
This is why the best approach is to create a new object instead of updating the existing one.

Reply:
This code was refactored to the wallet saga with a more appropriate explanation on https://github.com/HathorNetwork/hathor-wallet/pull/587/commits/6b87012bfa353140c7a864a832f81fdc097a799d